File: Regular Expressions/Code/chapter6/6-11/recipe6-11.py
#!/usr/bin/python
import re
from os import popen

output = popen( 'netstat -p tcp' )
regex = re.compile( r'^(?:.*\s+)([-\w.:]+)\s+([-\w.:]+)(?:\s+ESTABLISHED)$' )

lines = output.readlines()

for line in lines:
	if regex.match( line ):
		formatted = regex.sub( r'Found established connection:  \1 -> \2', line )
		print formatted,

output.close()

File: Regular Expressions/Code/chapter6/6-11/recipe6-11.php
<html>
<head><title>6.11 Filtering the output of netstat</title></head>
<body>
<form action="recipe6-11.php" method="post">
<input type="submit" value="Run netstat" /><br /><br />
<?php
if ( $_SERVER['REQUEST_METHOD'] == "POST" ) 
{
	$ps = popen( 'netstat -p tcp', 'r' );
	while ( $stuff = fgets( $ps, 2056 ) )
	{
		$output .= $stuff;
	}

	$lines = explode( "\n", $output );

	foreach ( $lines as $line )
	{
		print $line . "<br/>";
		if ( preg_match( "/^(?:.*\s+)([-\w.:]+)\s+([-\w.:]+)(?:\s+ESTABLISHED)$/", $line ) )
		{
			$newstr = preg_replace( "/^(?:.*\s+)([-\w.:]+)\s+([-\w.:]+)(?:\s+ESTABLISHED)$/", "Found established connection:  $1 -> $2", $line );
			print "<b>$newstr</b><br/>";
		}
	}

	pclose($ps);
}
?>
</form>
</body>
</html>
